src/GRA.Controllers/MissionControl/EmailManagementController.cs

```diff
@@ -207,5 +227,97 @@ public async Task<IActionResult> SendEmail(EmailIndexViewModel viewModel)
                 });
             }
         }
+
+        public async Task<IActionResult> Addresses()
+        {
+            var user = await _userService.GetDetails(GetActiveUserId());
+            if (user.IsAdmin)
+            {
+                var signupData = _emailManagementService.GetSignUpSourcesWithCount();
+                var viewModel = new EmailAddressesViewModel
+                {
+                    IsAdmin = user.IsAdmin,
+                    SignUpSources = new SelectList(signupData, "Value", "DisplayText")
+                };
+                return View(viewModel);
+            }
+            ShowAlertDanger("Page not found.");
+            return RedirectToAction(nameof(EmailManagementController.Index));
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> ExportAddresses(EmailAddressesViewModel viewModel)
+        {
+            var user = await _userService.GetDetails(GetActiveUserId());
+            if (user.IsAdmin)
+            {
+                if (string.IsNullOrEmpty(viewModel.SignUpSource))
+                {
+                    ShowAlertDanger("Could not export Email Reminders");
+                    ModelState.AddModelError(nameof(viewModel.SignUpSource), "The signup source is required.");
+                    var signupData = _emailManagementService.GetSignUpSourcesWithCount();
+                    viewModel.IsAdmin = user.IsAdmin;
+                    viewModel.SignUpSources = new SelectList(signupData, "Value", "DisplayText");
+                    return View("Addresses",viewModel);
+                }
+                var ms = new MemoryStream();
+                var writer = new StreamWriter(ms);
```

Comment:
This may also need to be in a `using()`. Anything that implements `IDisposable` needs to be wrapped in `using()`.